[User Friending] Improved Friends List UI with Cards

davdma opened the pull request:
Closes: #886 

Summary: The friends list UI is currently very bare bones, it shows just the email which is a hyperlink that leads to that friend's profile. However, we want something more user friendly. This PR goes to improve the template so that it fits in style wise with the rest of the profile UI built out right now.

What was added:

- Removed the `FriendTable` which is not a great UI for displaying friends
- Changed the template `userprofile_detail.html` to not display name if blank which addressed #1130
- Changed the template `user_friend_list.html` to use cards instead of `FriendTable`
- Cards include profile photo, username, name information for each friend
- Email information removed from friend list UI
- Styling added to template `user_friend_list.html` along with links to each friends profile through buttons

davdma committed:
[users] Updated profile template to not show name when empty

Fixed: #1130

davdma committed:
Merge branch 'dev' into users/update-ui

davdma committed:
[users] WPI: Removed FriendTable, created card template for friend list

WPI: #886

davdma committed:
Merge branch 'dev' into users/update-ui

davdma committed:
[users] Improved UI for friends list now in card form.

Friends list now opens to a page with cards for each friend with profile photo,
name, and username displayed. No email is displayed. A button links to each
friends profile.

Closes: #886

harriet-fisher reviewed (approved):
Nice update! Swapping out FriendsTable for cards is a big improvement; it’s way cleaner and more user-friendly. The context update makes things easier to read, and I like the extra check for missing names. Solid cleanup all around!! Thanks David

jonathanbacon22 commented on src/templates/users/user_friend_list.html:
```diff
@@ -1,7 +1,70 @@
+                     class="profile-photo rounded-circle me-3" />
+              {% else %}
+                <img src="{% static 'images/no_picture_available.png' %}"
+                     alt="Default Profile Photo"
```
Great!

jonathanbacon22 reviewed (approved):
Great changes! Thank you for adding this David. Are you aware of why the tests are failing?

davdma committed:
Merge branch 'dev' into users/update-ui

davdma commented:
Finished friends list now looks like this:

![image](https://github.com/user-attachments/assets/98a95858-653f-4053-9a68-e020d7b1e5f8)

davdma committed:
[users] Finished friends list UI styling

Closes: #886

hossainmd-uc reviewed (approved):
Nice job, the UI looks really clean. One question I have is what if a user has 1000 friends or more? Does the UI adapt to handle that many friends? Additionally, the title section looks a little scarce. A small section with "Friends" or some sort of indication would be a good addition for next time.

davdma commented:
@hossainmd-uc the html template can handle as many user cards as you want - the cards are simply formatted in rows with 3 columns, it can possibly keep going down forever. I can definitely add a My Friends title above in a later PR.

hossainmd-uc commented:
> @hossainmd-uc the html template can handle as many user cards as you want - the cards are simply formatted in rows with 3 columns, it can possibly keep going down forever. I can definitely add a My Friends title above in a later PR.

I guess my comment was more of a organization thing. like would it be neater to have like a button for "Next" instead of columns that extend infinitely. I think it might make more sense but we can definitely think about that. 

Also we can potentially have a search feature on this page directly for friends and it can filter by name or something

davdma commented:
@hossainmd-uc actually html does not automatically do scrolling - I'll need to make some tweaks. Thanks for raising the issue!

davdma committed:
[users] Fixed: correct redirect to user profile using user pk.

Originally links to profile was using profile pk. Then checked views
and realized that the profile view function took in the user pk of the profile.
Updated to ensure this.

davdma committed:
Merge branch 'dev' into users/update-ui

cmarshall282 reviewed (approved):
Great UI update. It looks nice in the screenshot you sent me!

cmarshall282 committed:
Merge branch 'dev' into users/update-ui

cmarshall282 commented:
Issue Score: Excellent

Comments:
Great job!

<!-- {"score": "excellent", "minor": false} -->
